proto/v3/message_contents/public_key.proto

```diff
@@ -34,6 +34,17 @@ message VmacAccountLinkedKey {
     }
 }
 
+// Keys linked to an account (e.g. signed by a wallet)
+// The purposes of the keys are encoded in the signatures
+message VmacAccountLinkedKeys {
+    VmacUnsignedPublicKey curve25519_key = 1;
+    VmacUnsignedPublicKey ed25519_key = 2;
```

Comment:
A `VmacUnsignedPublicKey` only has `curve25519` as an option for the key bytes, but doesn't have the option for `ed25519`. The field is just `bytes`, so we can shove anything we want in there, but we should allow it to mark the keys as being of different types in that message to avoid confusion. Otherwise you are going to access `key.ed25519_key.curve25519`, which is misleading and confusing.